hedera-node/hedera-smart-contract-service-impl/src/main/java/com/hedera/node/app/service/contract/impl/exec/metrics/ContractMetrics.java

```diff
@@ -0,0 +1,233 @@
+                final var config = new Counter.Config(METRIC_CATEGORY, name)
+                        .withDescription(descr)
+                        .withUnit(METRIC_TXN_UNIT);
+                final var metric = newCounter(metrics, config);
+                rejectedTxsLackingIntrinsicGas.put(txKind, metric);
+            }
+
+            // Rejected transactions for ethereum calls that are in type 3 blob transaction format
+            {
+                final var name = toRejectedName(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
+                final var descr = toRejectedDescr(REJECTED_TYPE3_FUNCTIONALITY, REJECTED_TYPE3_SHORT_DESCR);
+                final var config = new Counter.Config(METRIC_CATEGORY, name)
+                        .withDescription(descr)
+                        .withUnit(METRIC_TXN_UNIT);
+                final var metric = newCounter(metrics, config);
+                rejectedEthType3Counter = metric;
+            }
+        }
+
+        if (p2MetricsEnabled) {
+            // PLACEHOLDER
+        }
+    }
+
+    public void incrementRejectedTx(@NonNull final HederaFunctionality txKind) {
+        bumpRejectedTx(txKind, 1);
+    }
+
+    public void bumpRejectedTx(@NonNull final HederaFunctionality txKind, final long bumpBy) {
+        if (p1MetricsEnabled) requireNonNull(rejectedTxsCounters.get(txKind)).add(bumpBy);
+    }
+
+    public void incrementRejectedForGasTx(@NonNull final HederaFunctionality txKind) {
+        bumpRejectedForGasTx(txKind, 1);
+    }
+
+    public void bumpRejectedForGasTx(@NonNull final HederaFunctionality txKind, final long bumpBy) {
+        if (p1MetricsEnabled)
+            requireNonNull(rejectedTxsLackingIntrinsicGas.get(txKind)).add(bumpBy);
+    }
+
+    public void incrementRejectedType3EthTx() {
+        bumpRejectedType3EthTx(1);
+    }
+
+    public void bumpRejectedType3EthTx(final long bumpBy) {
+        if (p1MetricsEnabled) rejectedEthType3Counter.add(bumpBy);
+    }
+
+    public @NonNull Map<String, Long> getAllCounters() {
+        return Stream.concat(
+                        Stream.concat(
+                                rejectedTxsCounters.values().stream(),
+                                rejectedTxsLackingIntrinsicGas.values().stream()),
+                        Stream.ofNullable(rejectedEthType3Counter))
+                .collect(toMap(Counter::getName, Counter::get));
+    }
+
+    public @NonNull List<String> getAllCounterNames() {
+        return Stream.concat(
+                        Stream.concat(
+                                rejectedTxsCounters.values().stream(),
+                                rejectedTxsLackingIntrinsicGas.values().stream()),
+                        Stream.ofNullable(rejectedEthType3Counter))
+                .map(Metric::getName)
+                .sorted()
+                .toList();
+    }
+
+    public @NonNull List<String> getAllCounterDescriptions() {
```

Comment:
I guess these codecov is complaining about usage as these two methods do not have unit tests?

Reply:
Well, they don't have unit tests because they're only meant to be used in unit tests.  WIll annotate thusly.